accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,267 @@
+   - Collect organization admin credentials (user+password) on the HUB, and use the existing API. The downside is that it's more fields to be collected and managed.
+
+### Images
+
+For images metadata and pillar information are saved on the database. Images are store on disk but accessible through http(s). Images are associated to a Organization but it's IS is unique across installation.
+
+To copy data from one SUSE Manager server to another we would need to use several API calls and http(s) calls. Some data cannot be imported using the API, like package list information.
+
+An implementation example is available from the community [link](https://github.com/uyuni-project/contrib/blob/main/os-image-tools/osimage-import-export.py).
+
+Communication can be done from the HUB to the peripheral server to synchronize all the necessary data.
+
+### Activation Keys
+
+Activation keys are part of a organization and need to be mapped to peripherals at a organization mapping level. Similar to Configuration Channels we have API methods to create and update activation keys (namespace `activationkey`). However the API methods doesn't have the ability to do cross organization mapping.
+
+The same two options are available: New API endpoints or collect organization administrator credentials.
+
+## Synchronization failing scenarios
+
+We will bi-directional communication between Hub server and peripheral servers:
+
+1. **Peripheral to HUB:** Download packages when running `spacewalk-repo-sync`
+2. **HUB to Peripheral:** API call(s) to create software channels structure and in the future synchronize configuration channels, images and activation keys.
+
+Both synchronizations should happen automatically. For 1) it's already automated in taskomatic with `mgr-sync-refresh-default` task. for scenario 2) we would need to develop a new taskomatic task running on HUB server.
+
+If a peripheral server is unreachable the new taskomatic task should retry to apply the desired configurations in the next execution.
+
+In case HUB server is unreachable during repository sync, the next execution of the task should retry the synchonization and should be able to recover from the problem as soon HUB server gets available again.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Full synchronize a high number of peripherals at same time can overload the HUB server
+  - After the first full synchronization it should not be a problem
+  - One Peripheral can manage up to 10.000 minions. If we add 10 peripherals we would be able to manage 100.000 minions (more than our biggest customer)
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Create a new UI and use ISSv2 to synchronize data
+- Solve the existing and known problems of ISSv2
```

Comment:
I'd not discard this point, in addition to implement ISSv3.
As ISSv2 it's going to be used for disconnected environments, we can still bring a better user experience to that use case. For example, can we consider some improvement around performing parallel sql queries?

Reply:
In terms of performance, the main issues are in the import process. In there we cannot change it to run in parallel. 
We can however change the way we do the transaction, and have one transaction per channel, instead of one transaction per export as we have now.
Another possibility is to not have a transition at all, but that can be risky if users start to use the channel during the import process, or if an error occurs during the import.